back/app/policies/group_policy.rb

```diff
@@ -18,7 +18,21 @@ def resolve
     private
 
     def resolve_for_active
-      user.admin? ? scope.all : scope.none
+      return scope.all if user.admin?
+
+      moderated_projects = ::Project.where(id: user.moderatable_project_ids)
```

Comment:
I think this is a bug (that was already there). If you're a project moderator, you can see groups linked to it. But if you're a folder moderator, you cannot see groups linked to one of its projects.